### PR TITLE
Upgrade to support a Kubernetes nginx deployment and run as non-root

### PR DESCRIPTION
**Pull request by trnubo**
Includes
* Container runs as `nginx` user
* Nginx logs to stdout/stderr
* Nginx port defaults to 8080 and is configurable with PORT env
* Add README notes about changes
* Update examples
* Add k8s-init and k8s-nginx entrypoints, some entry.sh rework
* Add k8s-init.sh script to perform templating and copy to a volume
* Add support to templater.sh to update path when given path is not relative to the nginx root
* Add k8s tests
* Add test to check env-config.js templating
* Switch test to use bats-asset help functions

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ RUN yarn install --pure-lockfile --ignore-optional --no-progress
 FROM panubo/staticsite:latest
```
Comment (macropin) [inline docs/README.md]: use quay.io

File: docs/example/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN yarn install --pure-lockfile --ignore-optional --no-progress
 FROM panubo/staticsite:latest
```
Comment (macropin) [inline docs/example/Dockerfile]: also use quay.io

File: k8s-init.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# This script will copy the nginx config and html content to /volume then run
+# the normal renderers. This is used in an initContainer in a Kubernetes pod.
+# The /volume mount should then be mounted into the main container as
```
Comment (macropin) [inline k8s-init.sh:5]: /volume configurable?

Comment (trnubo) [reply k8s-init.sh:5]: Updated to make configurable

**Review by macropin — APPROVED** [review]
LGTM

**Pull request merged**